staging/src/k8s.io/client-go/testing/fixture.go

```diff
@@ -102,10 +102,24 @@ func ObjectReaction(tracker ObjectTracker) ReactionFunc {
 			if action.GetSubresource() == "" {
 				err = tracker.Create(gvr, action.GetObject(), ns)
 			} else {
-				// TODO: Currently we're handling subresource creation as an update
-				// on the enclosing resource. This works for some subresources but
-				// might not be generic enough.
-				err = tracker.Update(gvr, action.GetObject(), ns)
+				oldObj, getOldObjErr := tracker.Get(gvr, ns, objMeta.GetName())
+				if getOldObjErr != nil {
+					return true, nil, err
+				}
+				// Describe: Check whether the existing historical object type is the same as the current operation object type that needs to be updated, and if it is the same, perform the update operation.
+				// Purpose: The purpose of this is to prevent unexpected problems with fake client calls.
+				// Scenario: When a pod is expelled, if the tracker.Update operation is performed when the history and current object types are different, the following exception will occur when fake performs the pod list operation.
+				// Error：item[index]: can't assign or convert v1beta1.Eviction into v1.Pod.
```

Comment:
drop these pod-specific comment lines, the description is sufficient

Reply:
Received, resubmitted, and rebase